tarogarden/caretaker.go

```diff
@@ -323,10 +323,33 @@ func (b *BatchCaretaker) seedlingsToAssetSprouts(ctx context.Context,
 		}
 
 		var groupKey *asset.GroupKey
+		// If the seedling has a group key specified and emission
```

Comment:
In what cases could this happen, after a restart?

Reply:
So the intent was to validate the group key before ever writing the seedling to disk. AFAICT those earliest writes are in the planter in `prepTaroSeedling()`, so the earliest validation is here:

https://github.com/lightninglabs/taro/blob/69a66e45cabbb8d28ffc5526905cecdaebc24ac3/tarogarden/planter.go#L460

I think that this verification may then be redundant, and we just need this branch to use the specified key instead of creating a new one.

Reply:
Well if there's an incomplete batch, then we only have seedlings that haven't yet been mapped to "real" assets.